internal/jennies/golang/builder.go

```diff
@@ -90,105 +95,49 @@ func (jenny *Builder) generateBuilder(context languages.Context, builder ast.Bui
+				return jenny.emptyValueForGuard(context, guard.EmptyValueType)
 			},
 		}).
-		RenderAsBytes("builders/builder.tmpl", template.Builder{
-			Package:              builder.Package,
-			BuilderSignatureType: buildObjectSignature,
-			Imports:              imports,
-			BuilderName:          tools.UpperCamelCase(builder.Name),
-			ObjectName:           fullObjectName,
-			Comments:             builder.For.Comments,
-			Constructor:          builder.Constructor,
-			Properties:           builder.Properties,
-			Defaults:             jenny.genDefaultOptionsCalls(context, builder),
-			Options:              builder.Options,
-		})
-}
-
-func (jenny *Builder) genDefaultOptionsCalls(context languages.Context, builder ast.Builder) []template.OptionCall {
-	calls := make([]template.OptionCall, 0)
-	for _, opt := range builder.Options {
-		if opt.Default == nil {
-			continue
-		}
-
-		if len(opt.Args) == 0 {
-			continue
-		}
-
-		if hasTypedDefaults(opt) {
-			calls = append(calls, template.OptionCall{
-				OptionName: opt.Name,
-				Args:       jenny.formatDefaultTypedArgs(context, opt),
-			})
-			continue
-		}
-
-		calls = append(calls, template.OptionCall{
-			OptionName: opt.Name,
-			Args:       tools.Map(opt.Default.ArgsValues, formatScalar),
+		RenderAsBytes("builders/builder.tmpl", map[string]any{
+			"Package":              builder.Package,
+			"BuilderSignatureType": buildObjectSignature,
+			"Imports":              imports,
+			"BuilderName":          tools.UpperCamelCase(builder.Name),
+			"ObjectName":           fullObjectName,
+			"Comments":             builder.For.Comments,
+			"ConstructorName":      constructorName,
+			"Constructor":          builder.Constructor,
+			"Properties":           builder.Properties,
+			"Options":              builder.Options,
 		})
-	}
-
-	return calls
 }
 
-func hasTypedDefaults(opt ast.Option) bool {
-	for _, defArg := range opt.Default.ArgsValues {
-		if _, ok := defArg.(map[string]any); ok {
-			return true
-		}
-	}
-
-	return false
-}
+func (jenny *Builder) emptyValueForGuard(context languages.Context, typeDef ast.Type) string {
+	typeDef = typeDef.DeepCopy()
+	typeDef.Nullable = false
 
-func (jenny *Builder) formatDefaultTypedArgs(context languages.Context, opt ast.Option) []string {
-	args := make([]string, 0)
-	for i, arg := range opt.Default.ArgsValues {
-		val, _ := arg.(map[string]interface{})
-
-		pkg := ""
-		refPkg := ""
-		if opt.Args[i].Type.IsRef() {
-			refPkg = jenny.typeImportMapper(opt.Args[i].Type.AsRef().ReferredPkg)
-			pkg = opt.Args[i].Type.AsRef().ReferredType
-			_, isBuilder := context.Builders.LocateByObject(opt.Args[i].Type.AsRef().ReferredPkg, pkg)
-			obj, ok := context.LocateObject(opt.Args[i].Type.AsRef().ReferredPkg, pkg)
-			if !ok {
-				return []string{"unknown"}
+	switch typeDef.Kind {
+	case ast.KindRef:
+		resolvedType := context.ResolveRefs(typeDef)
+		if resolvedType.IsStruct() {
+			constructor := fmt.Sprintf("New%s()", typeDef.Ref.ReferredType)
```

Comment:
Nit: ditto using `+` op
```suggestion
			constructor := "New" + typeDef.Ref.ReferredType + "()"
```